packages/x-components/tests/e2e/history-queries/history-queries.spec.ts

```diff
@@ -46,6 +46,7 @@ And('a {string} of queries already searched', (list: string) => {
 
 When('history query number {int} is clicked', (historyQueryItem: number) => {
   cy.getByDataTest('history-query').eq(historyQueryItem).click().invoke('text').as('searchedQuery');
+  cy.waitForResultsToRender();
```

Comment:
Here I would say the same that I commented about the `query-suggestions` test.
I would add the step to wait for the results after the `    And   history query number <historyQueryItem> is clicked` of the first scenario:

```suggestion
```
```
    Then  related results are displayed
```